xarray/core/dataset.py

```diff
@@ -3519,39 +3519,97 @@ def _assert_all_in_dataset(
                 "cannot be found in this dataset"
             )
 
-    # Drop variables
-    @overload  # noqa: F811
-    def drop(
-        self, labels: Union[Hashable, Iterable[Hashable]], *, errors: str = "raise"
+    def drop_vars(
+        self, names: Union[Hashable, Iterable[Hashable]], *, errors: str = "raise"
     ) -> "Dataset":
-        ...
+        """Drop variables from this dataset.
 
-    # Drop index labels along dimension
-    @overload  # noqa: F811
-    def drop(
-        self, labels: Any, dim: Hashable, *, errors: str = "raise"  # array-like
-    ) -> "Dataset":
-        ...
+        Parameters
+        ----------
+        names : hashable or iterable of hashables
+            Name(s) of variables to drop.
+        errors: {'raise', 'ignore'}, optional
+            If 'raise' (default), raises a ValueError error if any of the variable
+            passed are not in the dataset. If 'ignore', any given names that are in the
+            dataset are dropped and no error is raised.
 
-    def drop(  # noqa: F811
-        self, labels=None, dim=None, *, errors="raise", **labels_kwargs
-    ):
-        """Drop variables or index labels from this dataset.
+        Returns
+        -------
+        dropped : Dataset
+
+        """
+        if isinstance(names, str) or not isinstance(names, Iterable):
```

Comment:
not sure if this should be `isinstance(names, str)` or `is_scalar(names)`?

Reply:
Good spot; I've changed this. But had to keep `or not isinstance(names, Iterable):` for mypy

@crusaderky as the resident expert, lmk if you have any thoughts on this.